File: src/main/java/com/example/BookStore/model/Author.java
package com.example.BookStore.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "Author")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Author {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
     private Long id;
    private String name;
   private String  contactDetails;

    @ManyToMany(mappedBy = "authors")
    private Set<Book> books;

    @ManyToOne
    @JoinColumn(name = "publisher_id")
    private Publisher publisher;
}

File: src/main/java/com/example/BookStore/model/Author.kt
package com.example.BookStore.model

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.*

@Entity
@Table(name = "Author")
data class Author(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long,
    var name: String,
    var contactDetails: String?,

    @ManyToMany(mappedBy = "authors")
    var books: MutableSet<Book>? = mutableSetOf<Book>(),

    @ManyToOne
    @JoinColumn(name = "publisher_id")
    var publisher: Publisher?



)
